purchases/src/main/kotlin/com/revenuecat/purchases/PostReceiptHelper.kt

```diff
@@ -0,0 +1,143 @@
+    private val deviceCache: DeviceCache,
+    private val subscriberAttributesManager: SubscriberAttributesManager
+) {
+    private val finishTransactions: Boolean
+        get() = appConfig.finishTransactions
+
+    /**
+     * This method will post a token and receiptInfo to the backend without consuming any purchases.
+     * It will store that the token was sent to the backend so it doesn't send it again.
+     */
+    fun postTokenAndReceiptInfoToBackend(
+        purchaseToken: String,
+        storeUserID: String?,
+        receiptInfo: ReceiptInfo,
+        isRestore: Boolean,
+        appUserID: String,
+        marketplace: String?,
+        onSuccess: () -> Unit,
+        onError: (PurchasesError) -> Unit,
+    ) {
+        syncSubscriberAttributesAndPostReceipt(
+            appUserID,
+            purchaseToken,
+            isRestore,
+            receiptInfo,
+            storeUserID,
+            marketplace,
+            onSuccess = { _, _ ->
+                deviceCache.addSuccessfullyPostedToken(purchaseToken)
+                onSuccess()
+            },
+            onError = { error, shouldConsumePurchase, _, _ ->
+                if (shouldConsumePurchase) {
+                    deviceCache.addSuccessfullyPostedToken(purchaseToken)
+                }
+                onError(error)
+            }
+        )
+    }
+
+    /**
+     * This method will post a StoreTransaction and optionally a StoreProduct info to the backend.
+     * It will consume the purchase if finishTransactions is true.
+     */
+    fun postTransactionToBackend(
+        purchase: StoreTransaction,
+        storeProduct: StoreProduct?,
+        isRestore: Boolean,
+        appUserID: String,
+        onSuccess: (SuccessfulPurchaseCallback)? = null,
+        onError: (ErrorPurchaseCallback)? = null
+    ) {
+        val receiptInfo = ReceiptInfo(
+            productIDs = purchase.productIds,
+            offeringIdentifier = purchase.presentedOfferingIdentifier,
+            storeProduct = storeProduct,
+            subscriptionOptionId = purchase.subscriptionOptionId
+        )
+        syncSubscriberAttributesAndPostReceipt(
+            appUserID = appUserID,
+            purchaseToken = purchase.purchaseToken,
+            isRestore = isRestore,
+            receiptInfo = receiptInfo,
+            storeUserID = purchase.storeUserID,
+            marketplace = purchase.marketplace,
+            onSuccess = { info, _ ->
+                billing.consumeAndSave(finishTransactions, purchase)
+                onSuccess?.let { it(purchase, info) }
+            },
+            onError = { error, shouldConsumePurchase, _, _ ->
+                if (shouldConsumePurchase) {
+                    billing.consumeAndSave(finishTransactions, purchase)
+                }
+                onError?.let { it(purchase, error) }
+            }
+        )
+    }
+
+    private fun syncSubscriberAttributesAndPostReceipt(
```

Comment:
I'd name it the other way around - the important part of this method is posting the receipt, posting attributes is secondary. 
So I'd go with `postReceiptAndSyncSubscriberAttributes`. 
Or maybe better `postReceiptAndSubscriberAttributes`, since having `sync` as a verb suggests a different action takes place, but it's all a part of the same sync